python/ray/serve/tests/test_api.py

```diff
@@ -884,6 +880,119 @@ async def __call__(self, request: Request) -> Dict[str, str]:
     }
 
 
+def test_get_app_handle_basic(serve_instance):
```

Comment:
Could you add a unit test to check that no code gets deserialized when `get_handle` is called? One way to do check if a deployment got deserialized is to create an actor that controls its serializability:

```python
@ray.remote
class SerializationController:

    def __init__(self):
        self._serialization_allowed: bool = True

    def serialization_allowed(self) -> bool:
        return self._serialization_allowed

    def disallow_serialization(self) -> bool:
        self._serialization_allowed = False

    def allow_serialization(self) -> bool:
        self._serialization_allowed = True

serialization_controller = SerializationController.remote()

@serve.deployment
class C:
    ...
    def __reduce__(self):
        if ray.get(serialization_controller.serialization_allowed.remote()):
            return ...  # Fill in the usual __reduce__ syntax here
        else:
            raise RuntimeError("Attempted to serialize!")
```